framework/decode/openxr_feature_util.cpp

```diff
@@ -0,0 +1,147 @@
+
+#include "util/logging.h"
+#include "util/platform.h"
+
+#include <cassert>
+#include <set>
+#include <string>
+#include <algorithm>
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(decode)
+GFXRECON_BEGIN_NAMESPACE(feature_util)
+
+XrResult GetApiLayers(PFN_xrEnumerateApiLayerProperties enum_layer_pros, std::vector<XrApiLayerProperties>* layers)
+{
+    assert(layers != nullptr);
+
+    XrResult result = XR_ERROR_INITIALIZATION_FAILED;
+
+    if (enum_layer_pros != nullptr)
+    {
+        uint32_t layer_count = 0;
+        result               = enum_layer_pros(layer_count, nullptr, nullptr);
+
+        if ((result == XR_SUCCESS) && (layer_count > 0))
+        {
+            layers->resize(layer_count);
+            for (uint32_t iii = 0; iii < layer_count; ++iii)
+            {
+                (*layers)[iii].type = XR_TYPE_API_LAYER_PROPERTIES;
+                (*layers)[iii].next = nullptr;
+            }
+            result = enum_layer_pros(static_cast<uint32_t>(layers->size()), &layer_count, layers->data());
+        }
+    }
+
+    return result;
+}
+
+bool IsSupportedApiLayer(const std::vector<XrApiLayerProperties>& properties, const char* layer)
+{
+    assert(layer != nullptr);
+
+    for (const auto& property : properties)
+    {
+        if (strcmp(property.layerName, layer) == 0)
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+XrResult GetInstanceExtensions(PFN_xrEnumerateInstanceExtensionProperties enum_ext_props,
+                               std::vector<XrExtensionProperties>*        properties)
+{
+    assert(properties != nullptr);
+
+    XrResult result = XR_ERROR_INITIALIZATION_FAILED;
+
+    if (enum_ext_props != nullptr)
+    {
+        uint32_t count = 0;
+        result         = enum_ext_props(nullptr, 0, &count, nullptr);
+
+        if ((result == XR_SUCCESS) && (count > 0))
+        {
+            properties->resize(count);
+            for (uint32_t iii = 0; iii < count; ++iii)
+            {
+                (*properties)[iii].type = XR_TYPE_EXTENSION_PROPERTIES;
+                (*properties)[iii].next = nullptr;
+            }
+            result = enum_ext_props(nullptr, static_cast<uint32_t>(properties->size()), &count, properties->data());
+        }
+    }
+
+    return result;
+}
+
+bool IsSupportedExtension(const std::vector<XrExtensionProperties>& properties, const char* extension)
+{
+    assert(extension != nullptr);
+
+    for (const auto& property : properties)
+    {
+        if (util::platform::StringCompare(property.extensionName, extension) == 0)
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void RemoveUnsupportedExtensions(const std::vector<XrExtensionProperties>& properties,
+                                 std::vector<const char*>*                 extensions)
+{
+    assert(extensions != nullptr);
+
+    auto extensionIter = extensions->begin();
+    while (extensionIter != extensions->end())
```

Comment:
This does not look like it would save us any code.  It would be roughly the same as what's here and this code was pulled from the Vulkan-side but re-purposed for OpenXR.  Plus we would have to re-define the IsSupportedExtension behavior above to work better with a non-iterator approach, and that function is being used beyond this internal case.  I think it's best to leave it as is for now.